pipeline_dp/budget_accounting.py

```diff
@@ -169,24 +187,56 @@ def _finalize(self):
             raise Exception("compute_budgets can not be called twice.")
         self._finalized = True
 
+    def _get_next_aggregation_index(self) -> int:
+        # todo: checks
+        self._next_aggregation_index += 1
+        return self._next_aggregation_index - 1
+
 
-@dataclass
 class BudgetAccountantScope:
 
-    def __init__(self, accountant, weight):
+    def __init__(self,
+                 accountant: BudgetAccountant,
+                 weight: float,
+                 index_aggregation_scope: Optional[int] = None):
         self.weight = weight
         self.accountant = accountant
+        self._index_aggregation_scope = index_aggregation_scope
+        self._epsilon = None
+        self._delta = None
+        if index_aggregation_scope is not None:
+            self._compute_budget_for_aggregation_scope()
```

Comment:
do we not set `self._index_aggregation_scope` after the computation? 
This is referenced in `_validate_epsilon_delta()`

Reply:
Now, it set 3 lines above